core/tx_noncer.go

```diff
@@ -45,43 +49,40 @@ func newTxNoncer(statedb *state.StateDB) *txNoncer {
 func (txn *txNoncer) get(addr common.Address) uint64 {
 	// We use mutex for get operation is the underlying
 	// state will mutate db even for read access.
-	txn.lock.Lock()
-	defer txn.lock.Unlock()
-
-	if _, ok := txn.nonces[addr]; !ok {
-		txn.nonces[addr] = txn.fallback.GetNonce(addr)
+	if _, ok := txn.nonces.Get(addr); !ok {
+		txn.lock.Lock()
+		txn.nonces.Add(addr, txn.fallback.GetNonce(addr))
+		txn.lock.Unlock()
```

Comment:
>  fixed-size thread safe LRU cache

```suggestion
		txn.nonces.Add(addr, txn.fallback.GetNonce(addr))
```
This should suffice, no?

Reply:
Or maybe simpler:
```golang
if nonce, ok := txn.nonces.Get(addr); ok {
     nonce.(uint64)
}
nonce := txn.fallback.GetNonce(addr)
txn.nonces.Add(addr,nonce)
return nonce
```

Reply:
I think method that return once would be good.